pkg/build/api/v1beta3/types.go

```diff
@@ -85,6 +85,9 @@ type BuildStatus struct {
 
 	// Config is an ObjectReference to the BuildConfig this Build is based on.
 	Config *kapi.ObjectReference `json:"config,omitempty"`
+
+	// Revision is the revision that was built with this build
+	Revision *SourceRevision `json:"revision,omitempty" description:"source revision built with this build"`
 }
```

Comment:
it really seems like a parallel `Revision *SourceRevision` status field to me that gets set to the resolved commit